variables/03-const.mdx

```diff
@@ -6,7 +6,11 @@ import { CodeSurfer } from "mdx-deck-code-surfer"
 
 - follows the same block scope rules as `let`
 - can not be re-assigned after it has a value
-- do not confuse this with immutability
+- do not confuse this with immutability*
+
+* Should this concept be introdued at this time?
+* Worth mentioning that `const` should be preferred over `let`, and usage of `let` should always be deliberate rather than wait-and-see?
```

Comment:
I think you are right in having a `let be the default` mindset and wait and see. 

I think using `const` for functions might throw people off. 


The reason why I think it's worth talking about immutability - although could find a different term (I know the concept of immutability made my head hurt a bit when first coming from C# to JavaScript) 

Is I've seen people expect that 

```js
const SOME_OBJ = {
x: 'value'
}
```

would ensure that SOME_OBJ.x could never be changed 

Can think of a more approachable way to explain this, as I could see terms like mutation/immutability/etc making it seem more intimidating than it actually is.

Reply:
Maybe it would be good to have an early block on "key differences" between C# and JS/TS, so you can introduce the concept there?

Reply:
Good idea --- the things that tend to trip people up the most is

* `this` 
* scope issues
* async behaviour 
* assuming that just because C# looks like TypeScript - it behaves the same 
* runtime time-checking vs build-time type checking
* JSON vs C# DynamicObjects / ExpandoObjects / Dictionary
* C# - Lambdas, Delegates, Linq - and how those concepts can relate to JavaScript/TypeScript

* [Good guide covering some of the basics of C# to TypeScript](https://cyrilletuzi.github.io/javascript-guides/csharp-to-typescript.html)
* [Microsoft Docs on C# Anon Functions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/anonymous-functions)
* [=> Operator Docs from MS](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-operator)
* [MS Guide on Delegates](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/delegates/)

Some of these things are probably going far deeper into what we need to cover - but I recall when I was last working with C#, simply trying to pass a function into a function was a pain, and the syntax for inline anonymous functions were a mess (seems like that's been cleaned up recently)

Reply:
The typescript docs also have a good guide on [this](https://www.typescriptlang.org/docs/handbook/functions.html#this) that we can probably leverage